src/components/AdminRegisterPage/index.jsx

```diff
@@ -19,18 +23,33 @@ const AdminRegisterPage = ({ match }) => {
     );
   }
 
-  const { enterpriseSlug } = match.params;
-  const isEnterpriseAdmin = isEnterpriseUser(user, ENTERPRISE_ADMIN);
-  if (isEnterpriseAdmin) {
-    // user is authenticated and has the ``enterprise_admin`` JWT role, so redirect user to
-    // account activation page to ensure they verify their email address.
-    return <Redirect to={`/${enterpriseSlug}/admin/register/activate`} />;
-  }
+  useEffect(() => {
+    const processEnterpriseAdmin = (enterpriseUUID) => {
+      const isEnterpriseAdmin = isEnterpriseUser(user, ENTERPRISE_ADMIN, enterpriseUUID);
+      if (isEnterpriseAdmin) {
+        // user is authenticated and has the ``enterprise_admin`` JWT role, so redirect user to
+        // account activation page to ensure they verify their email address.
+        history.push(`/${enterpriseSlug}/admin/register/activate`);
+      } else {
+        // user is authenticated but doesn't have the `enterprise_admin` JWT role; force a log out so their
+        // JWT roles gets refreshed. on their next login, the JWT roles will be updated.
+        const logoutRedirectUrl = getLogoutRedirectUrl(getProxyLoginUrl(enterpriseSlug));
+        global.location.href = logoutRedirectUrl;
+      }
+      return <EnterpriseAppSkeleton />;
+    };
+
+    LmsApiService.fetchEnterpriseBySlug(enterpriseSlug)
```

Comment:
just thinking, how about creating a custom hook and encapsulate this api call and stuff in that hook? 

something like `const [enterpriseUUID, fetchError] = useEnterpriseUUID(enterpriseSlug)`

[a example of custom hook](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/60694bad245bd54de8d227b135eb136d53395c4c/src/components/program-progress/data/hooks.jsx#L28)

and the logic of page can be cleaner

```
const [enterpriseUUID, fetchError] = useEnterpriseUUID(enterpriseSlug)

if (!enterpriseUUID) {
return Skeleton
} else {

// dump the logic of processEnterpriseAdmin
}

return Skeleton
```

I feel like it would look cleaner plus hook will be reusable.